crates/viewer/re_renderer/src/resource_managers/chroma_subsampling_converter.rs

```diff
@@ -0,0 +1,336 @@
+            gpu_data::UniformBuffer {
+                format: format as _,
+                primaries: primaries as _,
+                target_texture_size: [output_width, output_height],
+
+                _end_padding: Default::default(),
+            },
+        );
+
+        let bind_group = ctx.gpu_resources.bind_groups.alloc(
+            &ctx.device,
+            &ctx.gpu_resources,
+            &BindGroupDesc {
+                label: "RectangleInstance::bind_group".into(),
+                entries: smallvec![
+                    uniform_buffer,
+                    BindGroupEntry::DefaultTextureView(input_data.handle),
+                ],
+                layout: renderer.bind_group_layout,
+            },
+        );
+
+        Self {
+            bind_group,
+            target_texture,
+        }
+    }
+
+    /// Runs the conversion from the input texture data.
+    pub fn convert_input_data_to_texture(
+        self,
+        ctx: &RenderContext,
+    ) -> Result<GpuTexture, DrawError> {
+        // TODO(andreas): Does this have to be on the global view encoder?
+        // If this ever becomes a problem we could easily schedule this to another encoder as long as
+        // we guarantee that the conversion is enqueued before the resulting texture is used.
+        // Given that we already have this neatly encapsulated work package this would be quite easy to do!
+        let mut encoder = ctx.active_frame.before_view_builder_encoder.lock();
+        let mut pass = encoder
+            .get()
+            .begin_render_pass(&wgpu::RenderPassDescriptor {
+                label: self.target_texture.creation_desc.label.get(),
+                color_attachments: &[Some(wgpu::RenderPassColorAttachment {
+                    view: &self.target_texture.default_view,
+                    resolve_target: None,
+                    ops: wgpu::Operations {
+                        load: wgpu::LoadOp::Clear(wgpu::Color::BLACK),
+                        store: wgpu::StoreOp::Store,
+                    },
+                })],
+                ..Default::default()
+            });
+
+        ctx.renderer::<ChromaSubsamplingConverter>().draw(
+            &ctx.gpu_resources.render_pipelines.resources(),
+            crate::draw_phases::DrawPhase::Opaque, // Don't care about the phase.
+            &mut pass,
+            &self,
+        )?;
+
+        Ok(self.target_texture)
+    }
+}
+
+/// Converter for chroma subsampling formats.
+///
+/// Takes chroma subsampled data and draws to a fullscreen sRGB output texture.
+/// Implemented as a [`Renderer`] in order to make use of the existing mechanisms for storing renderer data.
+/// (we need some place to lazily create the render pipeline, store a handle to it and encapsulate the draw logic!)
+pub struct ChromaSubsamplingConverter {
+    render_pipeline: GpuRenderPipelineHandle,
+    bind_group_layout: GpuBindGroupLayoutHandle,
+}
+
+impl Renderer for ChromaSubsamplingConverter {
```

Comment:
Hmm not happy with this either because we don't upsample really as in changing the resolution, especially once YUV444 enters the picture. That said, I also don't like status quo. Maybe the whole family of things here should be `YuvFormatConverter` because it converts any yuv format that we want to handle?